docs/content/docs/operations/ingress.md

```diff
@@ -25,33 +25,79 @@ under the License.
 -->
 
 # Accessing Flink’s Web UI
-
 The Flink Kubernetes Operator, by default, does not change the way the native kubernetes integration [exposes](https://nightlies.apache.org/flink/flink-docs-master/docs/deployment/resource-providers/native_kubernetes/#accessing-flinks-web-ui) the Flink Web UI.
 
 # Ingress
-The Operator also supports creating an optional [Ingress](https://kubernetes.io/docs/concepts/services-networking/ingress/) entry for Web UI access. Ingress generation can be turned on by setting the `ingressDomain` field in the FlinkDeployment:
-
+Beyond the native options, the Operator also supports creating [Ingress](https://kubernetes.io/docs/concepts/services-networking/ingress/) entries for external UI access. 
+Ingress generation can be turned on by defining the `ingress` field in the `FlinkDeployment`:
 ```yaml
-apiVersion: flink.apache.org/v1alpha1
-kind: FlinkDeployment
 metadata:
   namespace: default
-  name: basic-ingress
+  name: advanced-ingress
 spec:
   image: flink:1.14.3
   flinkVersion: v1_14
-  ingressDomain: flink.k8s.io
-  ...
+  ingress:
+    template: "flink.k8s.io/{{namespace}}/{{name}}(/|$)(.*)"
+    className: "nginx"
+    annotations:
+      nginx.ingress.kubernetes.io/rewrite-target: "/$2"
+```
+The `ingress` specification defines a mandatory `template` field and two optional fields `className` and `annotations`. 
+When the CR is submitted the Operator substitutes the template variables from metadata and creates an Ingress entry on the Kubernetes cluster.  
+Given the example above the Flink UI could be accessed at https://flink.k8s.io/default/advanced-ingress/ and the generated Ingress entry would be:
```

Comment:
Could we add a note about that the trailing slash is necessary? I am afraid some users will run into this issue.

Reply:
done